x-pack/plugins/ml/public/application/components/navigation_menu/main_tabs.tsx

```diff
@@ -87,16 +118,32 @@ export const MainTabs: FC<Props> = ({ tabId, disableLinks }) => {
   }
 
   const tabs = getTabs(disableLinks);
+  const mlUrlGenerator = useMlUrlGenerator();
+  const navigateToPath = useNavigateToPath();
+
+  const redirectToTab = async (defaultPathId: MlUrlGeneratorState['page']) => {
+    // TODO: fix ts
+    // @ts-ignore
+    const path = await mlUrlGenerator.createUrl({
+      page: defaultPathId,
+      pageState: { globalState },
+    });
+
+    await navigateToPath(path, false);
+  };
+
+  useEffect(() => {
+    document.title = `ML - ${TAB_DATA[selectedTabId].name} - Kibana`;
```

Comment:
I wonder if Kibana offers a way to do this more "officially" without directly manipulating `document.title`?

Reply:
That's a good point! I do see `coreStart.docTitle.change('My Title')` that the Kibana team has exposed out ... Gonna take a look to see if we an implement this using that instead 👍

Reply:
Changed to using `chrome.docTitle.change()` here https://github.com/elastic/kibana/pull/76735/commits/c62a924a86d160f7f5a8804fcf8d68285d3aef42